tests/test_use.py

```diff
@@ -28,3 +30,37 @@ def test_main():
     assert op2.results[0].uses == set()
 
     print("Done")
+
+
+test_prog_blocks = """
+func.func @blocks() {
+  "test.termop"() [^0, ^1] : () -> ()
+^0:
+  cf.br ^2
+^1:
+  cf.br ^2
+^2:
+  func.return
+}
+"""
+
+
+def test_predecessor():
+    ctx = MLContext()
+    ctx.load_dialect(Builtin)
+    ctx.load_dialect(Test)
+    ctx.load_dialect(Func)
+    ctx.load_dialect(Cf)
```

Comment:
nit:
it feels like it would be good to do it all with just the test dialect, without `func` and `cf`.